TACC/Frontera-Portal
# task/FP-790 Shared workspaces listing

## Overview: ##

Front end listing of shared workspaces (projects)

## PR Status: ##


## Related Jira tickets: ##

* [FP-213](https://jira.tacc.utexas.edu/browse/FP-213)
* [FP-790](https://jira.tacc.utexas.edu/browse/FP-790)

## Summary of Changes: ##

- Sagas for projects listing
- DataFilesProjectsList component (for listing all your projects)
- DataFilesProjectListing component (for listing files within a project, built around DataFilesListing)

## Testing Steps: ##
1. Set your projects prefix settings (and re-use CEP project root root)
```
_PORTAL_DATA_DEPOT_PROJECTS_SYSTEM_PREFIX = 'frontera.SOMETHINGUNIQUEdev.project'
_PORTAL_PROJECTS_NAME_PREFIX = _PORTAL_DATA_DEPOT_PROJECTS_SYSTEM_PREFIX
_PORTAL_PROJECTS_ID_PREFIX = _PORTAL_NAMESPACE.upper()
_PORTAL_PROJECTS_ROOT_HOST = 'data.tacc.utexas.edu'
_PORTAL_PROJECTS_ROOT_DIR = '/corral-repl/tacc/aci/CEP/projects'
_PORTAL_PROJECTS_ROOT_SYSTEM_NAME = 'cep.project.root
_PORTAL_PROJECTS_PRIVATE_KEY = (
  bunch of private key lines (multi-line format)
)
_PORTAL_PROJECTS_PUBLIC_KEY = (
  bunch of public key lines (multi-line format but probably doesn't matter as much)
)
```
2. Migrate the projects backend with 
```
python manage.py migrate
```
3. Open up a python shell
4. Set up the initial ProjectId model 
```
from portal.apps.projects.models.base import ProjectId
ProjectId.objects.create(value=1).save()
```
5. Create a project
```
from portal.apps.projects.managers.base import ProjectsManager
from django.contrib.auth import get_user_model

user = get_user_model().objects.get(username="USERNAME")
mgr = ProjectsManager(user)
prj = mgr.create('My Test Project')
prj.add_pi(user)
```
6. Check to see that you can see your project in the list of Shared Workspaces
7. Check to see that you can retrieve your files (I currently can't without pushing keys and i think this is wrong)


## UI Photos:

![image](https://user-images.githubusercontent.com/17181582/101214038-f3249580-3640-11eb-90e8-9b13684d095b.png)

## Notes: ##
